csrc/ops/arith.cpp

```diff
@@ -2358,8 +2358,12 @@ TensorView* fusedMultiplySum(
   // TODO:
   //  Add tf32 and other mma data types
   //  Add fallback path for non-mma data types.
-  TORCH_CHECK(tv_a->getDataType().value() == DataType::Half);
-  TORCH_CHECK(tv_b->getDataType().value() == DataType::Half);
+  TORCH_CHECK(
```

Comment:
Based on the added test, the goal is to support for problems where bf16 is the type of both inputs. The checks here will let the following mixed precision (`tv_a` -> bf16, `tv_b` -> fp16).

How about adding one more check:
```
TORCH_CHECK(tv_b->getDataType().value() == tv_b->getDataType().value() );
```
?

Reply:
changed